pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -986,27 +986,42 @@ func (p *PriorityQueue) Update(logger klog.Logger, oldPod, newPod *v1.Pod) error
 	if usPodInfo := p.unschedulablePods.get(newPod); usPodInfo != nil {
 		pInfo := updatePod(usPodInfo, newPod)
 		p.updateNominatedPodUnlocked(logger, oldPod, pInfo.PodInfo)
-		if isPodUpdated(oldPod, newPod) {
-			gated := usPodInfo.Gated
-			if p.isPodBackingoff(usPodInfo) {
-				if err := p.podBackoffQ.Add(pInfo); err != nil {
-					return err
-				}
-				p.unschedulablePods.delete(usPodInfo.Pod, gated)
-				logger.V(5).Info("Pod moved to an internal scheduling queue", "pod", klog.KObj(pInfo.Pod), "event", PodUpdate, "queue", backoffQ)
-			} else {
-				if added, err := p.addToActiveQ(logger, pInfo); !added {
-					return err
-				}
+		gated := usPodInfo.Gated
+		if p.isSchedulingQueueHintEnabled {
+			// When unscheduled Pods are updated, we check with QueueingHint
```

Comment:
Should we call isPodUpdated here and proceed with the requeueing logic only if true? would that give us another layer of protection against unnecessary requeueing?

Reply:
For now, yes we can do this. 
But for the future, there may be some plugins trying to make decision based on the change of pod's status (For example, https://github.com/kubernetes/enhancements/issues/3990 ?)
So I tend to not call `isPodUpdated` here and let the plugin's queueHint functions decide later.

Reply:
Agree with @AxeZhan. 
Regardless of how https://github.com/kubernetes/enhancements/issues/3990 goes, all filtering should be done with QHint and we shouldn't make any additional layer built into the scheduler, which doesn't allow users to customize.